exercises/chapter7/test/no-peeking/Solutions.purs

```diff
@@ -84,6 +85,29 @@ instance eqTree :: Eq a => Eq (Tree a) where
 instance showTree :: Show a => Show (Tree a) where
   show t = genericShow t
 
+{-
+-- Solution using manually-written instances:
+
+instance eqTree :: Eq a => Eq (Tree a) where
+  eq Leaf Leaf = true
+  eq (Branch t1a va t2a) (Branch t1b vb t2b) =
+    t1a == t1b
+      && va == vb
+      && t2a == t2b
```

Comment:
Another autoformatting struggle.

I ideally want something like:
```purs
  eq (Branch t1a va t2a) (Branch t1b vb t2b)
      =  t1a == t1b
      && va  == vb
      && t2a == t2b
```

These are the only purty-friendly options I could find:
```purs
  eq (Branch t1a va t2a) (Branch t1b vb t2b) =
    t1a == t1b
      && va
      == vb
      && t2a
      == t2
```
```purs
  eq (Branch t1a va t2a) (Branch t1b vb t2b) = t1a == t1b && va == vb && t2a == t2b
```
```purs
  -- Unnecessary parens
  eq (Branch t1a va t2a) (Branch t1b vb t2b) =
    (t1a == t1b)
      && (va == vb)
      && (t2a == t2b)
```